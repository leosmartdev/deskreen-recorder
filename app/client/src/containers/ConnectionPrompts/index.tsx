import React, { useContext } from 'react';
import { Row, Col } from 'react-flexbox-grid';
import { useTranslation } from 'react-i18next';
import { AppContext } from '../../providers/AppContextProvider';
import {
  DARK_UI_BACKGROUND,
  LIGHT_UI_BACKGROUND,
} from '../../constants/styleConstants';
import MyDeviceInfoCard from '../../components/MyDeviceInfoCard';
import { TFunction } from 'i18next';
import { H2, H3 } from '@blueprintjs/core';
import ConnectingIndicator from '../../components/ConnectingIndicator';

const Slide = require('react-reveal/Slide');

interface ConnectionPropmptsProps {
  myDeviceDetails: DeviceDetails;
  isShownTextPrompt: boolean;
  promptStep: number;
  connectionIconType: ConnectionIconType;
  isShownSpinnerIcon: boolean;
  spinnerIconType: LoadingSharingIconType;
  isOpen: boolean;
}

function getPromptContent(t: TFunction, step: number) {
  switch (step) {
    case 1:
      return (
        <H3>
          {t(
            'Waiting for user to click ALLOW button on screen sharing device...'
          )}
        </H3>
      );
    case 2:
      return <H3>{t('Connected!')}</H3>;
    case 3:
      return (
        <H3>
          {t(
            'Waiting for user to select source to share from screen sharing device...'
          )}
        </H3>
      );
    default:
      return <H3>{`${t('Error occurred')} :(`}</H3>;
  }
}

function ConnectionPropmpts(props: ConnectionPropmptsProps) {
  const {
    myDeviceDetails,
    promptStep,
    connectionIconType,
    isShownSpinnerIcon,
    spinnerIconType,
    isOpen,
  } = props;

  const { t } = useTranslation();
  const { isDarkTheme } = useContext(AppContext);

  return (
    <Slide id="connection-prompts-slide" bottom when={isOpen} duration={1000}>
      <div
        style={{
          position: 'absolute',
          zIndex: 10,
          top: 0,
          left: 0,
          width: '100%',
          height: '100vh',
          boxShadow: `0 0 0 5px ${isDarkTheme ? '#000' : '#A7B6C2'}`,
          backgroundColor: isDarkTheme
            ? DARK_UI_BACKGROUND
            : LIGHT_UI_BACKGROUND,
        }}
      >
        <Row
          bottom="xs"
          style={{
            height: '50vh',
            width: '100%',
            marginRight: '0px',
            marginLeft: '0px',
          }}
        >
          <Row center="xs" style={{ width: '100%', margin: '0 auto' }}>
            <Col
              xs={12}
              style={{
                marginBottom: '50px',
                textAlign: 'center',
                width: '100%',
              }}
            >
              <div style={{ width: '100%' }}>
                <Row center="xs">
                  <H2 className="bp3-text-muted">{t('Deskreen Screen Viewer')}</H2>
                </Row>
                <Row center="xs" style={{ width: '100%', margin: '0 auto' }}>
                  <Col md={6} xl={4}>
                    <MyDeviceInfoCard deviceDetails={myDeviceDetails} />
                  </Col>
                </Row>
                <div id="prompt-text" style={{ fontSize: '20px' }}>
                  {getPromptContent(t, promptStep)}
                </div>
              </div>
            </Col>
          </Row>
        </Row>
        <ConnectingIndicator
          currentStep={promptStep}
          connectionIconType={connectionIconType}
          isShownSelectingSharingIcon={isShownSpinnerIcon}
          selectingSharingIconType={spinnerIconType}
        />
      </div>
    </Slide>
  );
}

export default ConnectionPropmpts;
