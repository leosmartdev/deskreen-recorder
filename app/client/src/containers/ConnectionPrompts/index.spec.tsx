import React from 'react';
import renderer from 'react-test-renderer';
import ConnectionPrompts from '.';

jest.useFakeTimers();

const TEST_DEVICE_DETAILS = {
  myIP: '',
  myOS: '',
  myDeviceType: '',
  myBrowser: '',
};
describe('when getPromptContent is called', () => {
  it('should match exact snapshot on each step', () => {
    for (let i = 0; i < 4; ++i) {
      const subject = renderer.create(
        <>
          <ConnectionPrompts
            myDeviceDetails={TEST_DEVICE_DETAILS}
            isShownTextPrompt
            promptStep={i}
            connectionIconType="feed"
            isShownSpinnerIcon
            spinnerIconType="application"
            isOpen
          />
        </>
      );
      expect(subject).toMatchSnapshot();
    }
  });
});
