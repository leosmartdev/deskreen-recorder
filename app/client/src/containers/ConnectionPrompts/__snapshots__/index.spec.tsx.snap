// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when getPromptContent is called should match exact snapshot on each step 1`] = `
<div
  className="react-reveal"
  style={
    Object {
      "backgroundColor": "rgba(240, 248, 250, 1)",
      "boxShadow": "0 0 0 5px #A7B6C2",
      "height": "100vh",
      "left": 0,
      "opacity": undefined,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <div
    className="row bottom-xs"
    style={
      Object {
        "height": "50vh",
        "marginLeft": "0px",
        "marginRight": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row center-xs"
      style={
        Object {
          "margin": "0 auto",
          "width": "100%",
        }
      }
    >
      <div
        className="col-xs-12"
        style={
          Object {
            "marginBottom": "50px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="row center-xs"
          >
            <h2
              className="bp3-heading bp3-text-muted"
            >
              Deskreen Screen Viewer
            </h2>
          </div>
          <div
            className="row center-xs"
            style={
              Object {
                "margin": "0 auto",
                "width": "100%",
              }
            }
          >
            <div
              className="col-md-6 col-xl-4"
            >
              <div
                className="bp3-card bp3-elevation-3"
                style={
                  Object {
                    "backgroundColor": "rgba(240, 248, 250, 1)",
                    "marginBottom": "30px",
                  }
                }
              >
                <h3
                  className="bp3-heading"
                >
                  My Device Info:
                </h3>
                <div
                  className="bp3-callout"
                >
                  <div
                    className=""
                  >
                    Device Type: 
                  </div>
                  <span
                    className="bp3-popover-wrapper"
                  >
                    <span
                      className="bp3-popover-target"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <div
                        className=""
                        style={
                          Object {
                            "backgroundColor": "#00f99273",
                            "borderRadius": "20px",
                            "fontWeight": 900,
                            "paddingLeft": "10px",
                            "paddingRight": "10px",
                          }
                        }
                        tabIndex={0}
                      >
                        <div
                          className=""
                        >
                          Device IP: 
                        </div>
                      </div>
                    </span>
                  </span>
                  <div
                    className=""
                  >
                    Device Browser: 
                  </div>
                  <div
                    className=""
                  >
                    Device OS: 
                  </div>
                </div>
                <div
                  className="bp3-text-muted"
                >
                  These details should match with the ones that you see in alert popup on computer screen, where Deskreen is running
                </div>
              </div>
            </div>
          </div>
          <div
            id="prompt-text"
            style={
              Object {
                "fontSize": "20px",
              }
            }
          >
            <h3
              className="bp3-heading"
            >
              Error occurred :(
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row top-xs"
    id="connecting-screen"
    style={
      Object {
        "height": "50vh",
        "marginLeft": "0px",
        "marginRight": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className=""
    >
      Error occurred :(
    </div>
  </div>
</div>
`;

exports[`when getPromptContent is called should match exact snapshot on each step 2`] = `
<div
  className="react-reveal"
  style={
    Object {
      "backgroundColor": "rgba(240, 248, 250, 1)",
      "boxShadow": "0 0 0 5px #A7B6C2",
      "height": "100vh",
      "left": 0,
      "opacity": undefined,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <div
    className="row bottom-xs"
    style={
      Object {
        "height": "50vh",
        "marginLeft": "0px",
        "marginRight": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row center-xs"
      style={
        Object {
          "margin": "0 auto",
          "width": "100%",
        }
      }
    >
      <div
        className="col-xs-12"
        style={
          Object {
            "marginBottom": "50px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="row center-xs"
          >
            <h2
              className="bp3-heading bp3-text-muted"
            >
              Deskreen Screen Viewer
            </h2>
          </div>
          <div
            className="row center-xs"
            style={
              Object {
                "margin": "0 auto",
                "width": "100%",
              }
            }
          >
            <div
              className="col-md-6 col-xl-4"
            >
              <div
                className="bp3-card bp3-elevation-3"
                style={
                  Object {
                    "backgroundColor": "rgba(240, 248, 250, 1)",
                    "marginBottom": "30px",
                  }
                }
              >
                <h3
                  className="bp3-heading"
                >
                  My Device Info:
                </h3>
                <div
                  className="bp3-callout"
                >
                  <div
                    className=""
                  >
                    Device Type: 
                  </div>
                  <span
                    className="bp3-popover-wrapper"
                  >
                    <span
                      className="bp3-popover-target"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <div
                        className=""
                        style={
                          Object {
                            "backgroundColor": "#00f99273",
                            "borderRadius": "20px",
                            "fontWeight": 900,
                            "paddingLeft": "10px",
                            "paddingRight": "10px",
                          }
                        }
                        tabIndex={0}
                      >
                        <div
                          className=""
                        >
                          Device IP: 
                        </div>
                      </div>
                    </span>
                  </span>
                  <div
                    className=""
                  >
                    Device Browser: 
                  </div>
                  <div
                    className=""
                  >
                    Device OS: 
                  </div>
                </div>
                <div
                  className="bp3-text-muted"
                >
                  These details should match with the ones that you see in alert popup on computer screen, where Deskreen is running
                </div>
              </div>
            </div>
          </div>
          <div
            id="prompt-text"
            style={
              Object {
                "fontSize": "20px",
              }
            }
          >
            <h3
              className="bp3-heading"
            >
              Waiting for user to click ALLOW button on screen sharing device...
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row top-xs"
    id="connecting-screen"
    style={
      Object {
        "height": "50vh",
        "marginLeft": "0px",
        "marginRight": "0px",
        "width": "100%",
      }
    }
  >
    <div
      id="pulsing-circle-1"
      style={
        Object {
          "backgroundColor": "rgba(43, 149, 214, 0.7)",
          "borderRadius": "100%",
          "height": "100px",
          "left": "0",
          "marginLeft": "auto",
          "marginRight": "auto",
          "position": "absolute",
          "right": "0",
          "textAlign": "center",
          "width": "100px",
          "zIndex": 1,
        }
      }
    />
    <div
      id="pulsing-circle-2"
      style={
        Object {
          "backgroundColor": "#2B95D6",
          "borderRadius": "100%",
          "height": "100px",
          "left": "0",
          "marginLeft": "auto",
          "marginRight": "auto",
          "position": "absolute",
          "right": "0",
          "textAlign": "center",
          "width": "100px",
          "zIndex": 2,
        }
      }
    >
      <div
        className="row center-xs middle-xs"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          className=""
        >
          <span
            className="bp3-icon bp3-icon-feed"
            icon="feed"
            style={
              Object {
                "transform": "translateX(10px)",
              }
            }
          >
            <svg
              data-icon="feed"
              fill="white"
              height={50}
              viewBox="0 0 20 20"
              width={50}
            >
              <desc>
                feed
              </desc>
              <path
                d="M2.5 15a2.5 2.5 0 000 5 2.5 2.5 0 000-5zm.5-5c-.55 0-1 .45-1 1s.45 1 1 1c2.76 0 5 2.24 5 5 0 .55.45 1 1 1s1-.45 1-1c0-3.87-3.13-7-7-7zM3 0c-.55 0-1 .45-1 1s.45 1 1 1c8.28 0 15 6.72 15 15 0 .55.45 1 1 1s1-.45 1-1C20 7.61 12.39 0 3 0zm0 5c-.55 0-1 .45-1 1s.45 1 1 1c5.52 0 10 4.48 10 10 0 .55.45 1 1 1s1-.45 1-1C15 10.37 9.63 5 3 5z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when getPromptContent is called should match exact snapshot on each step 3`] = `
<div
  className="react-reveal"
  style={
    Object {
      "backgroundColor": "rgba(240, 248, 250, 1)",
      "boxShadow": "0 0 0 5px #A7B6C2",
      "height": "100vh",
      "left": 0,
      "opacity": undefined,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <div
    className="row bottom-xs"
    style={
      Object {
        "height": "50vh",
        "marginLeft": "0px",
        "marginRight": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row center-xs"
      style={
        Object {
          "margin": "0 auto",
          "width": "100%",
        }
      }
    >
      <div
        className="col-xs-12"
        style={
          Object {
            "marginBottom": "50px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="row center-xs"
          >
            <h2
              className="bp3-heading bp3-text-muted"
            >
              Deskreen Screen Viewer
            </h2>
          </div>
          <div
            className="row center-xs"
            style={
              Object {
                "margin": "0 auto",
                "width": "100%",
              }
            }
          >
            <div
              className="col-md-6 col-xl-4"
            >
              <div
                className="bp3-card bp3-elevation-3"
                style={
                  Object {
                    "backgroundColor": "rgba(240, 248, 250, 1)",
                    "marginBottom": "30px",
                  }
                }
              >
                <h3
                  className="bp3-heading"
                >
                  My Device Info:
                </h3>
                <div
                  className="bp3-callout"
                >
                  <div
                    className=""
                  >
                    Device Type: 
                  </div>
                  <span
                    className="bp3-popover-wrapper"
                  >
                    <span
                      className="bp3-popover-target"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <div
                        className=""
                        style={
                          Object {
                            "backgroundColor": "#00f99273",
                            "borderRadius": "20px",
                            "fontWeight": 900,
                            "paddingLeft": "10px",
                            "paddingRight": "10px",
                          }
                        }
                        tabIndex={0}
                      >
                        <div
                          className=""
                        >
                          Device IP: 
                        </div>
                      </div>
                    </span>
                  </span>
                  <div
                    className=""
                  >
                    Device Browser: 
                  </div>
                  <div
                    className=""
                  >
                    Device OS: 
                  </div>
                </div>
                <div
                  className="bp3-text-muted"
                >
                  These details should match with the ones that you see in alert popup on computer screen, where Deskreen is running
                </div>
              </div>
            </div>
          </div>
          <div
            id="prompt-text"
            style={
              Object {
                "fontSize": "20px",
              }
            }
          >
            <h3
              className="bp3-heading"
            >
              Connected!
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row top-xs"
    id="connecting-screen"
    style={
      Object {
        "height": "50vh",
        "marginLeft": "0px",
        "marginRight": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="pulse-3-once"
      id="pulsing-circle-3"
      style={
        Object {
          "backgroundColor": "#15B371",
          "borderRadius": "100%",
          "height": "100px",
          "left": "0",
          "marginLeft": "auto",
          "marginRight": "auto",
          "position": "absolute",
          "right": "0",
          "textAlign": "center",
          "width": "100px",
        }
      }
    >
      <div
        className="row center-xs middle-xs"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          className=""
        >
          <span
            className="bp3-icon bp3-icon-feed"
            icon="feed"
            style={
              Object {
                "transform": "translateX(10px)",
              }
            }
          >
            <svg
              data-icon="feed"
              fill="white"
              height={50}
              viewBox="0 0 20 20"
              width={50}
            >
              <desc>
                feed
              </desc>
              <path
                d="M2.5 15a2.5 2.5 0 000 5 2.5 2.5 0 000-5zm.5-5c-.55 0-1 .45-1 1s.45 1 1 1c2.76 0 5 2.24 5 5 0 .55.45 1 1 1s1-.45 1-1c0-3.87-3.13-7-7-7zM3 0c-.55 0-1 .45-1 1s.45 1 1 1c8.28 0 15 6.72 15 15 0 .55.45 1 1 1s1-.45 1-1C20 7.61 12.39 0 3 0zm0 5c-.55 0-1 .45-1 1s.45 1 1 1c5.52 0 10 4.48 10 10 0 .55.45 1 1 1s1-.45 1-1C15 10.37 9.63 5 3 5z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when getPromptContent is called should match exact snapshot on each step 4`] = `
<div
  className="react-reveal"
  style={
    Object {
      "backgroundColor": "rgba(240, 248, 250, 1)",
      "boxShadow": "0 0 0 5px #A7B6C2",
      "height": "100vh",
      "left": 0,
      "opacity": undefined,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <div
    className="row bottom-xs"
    style={
      Object {
        "height": "50vh",
        "marginLeft": "0px",
        "marginRight": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row center-xs"
      style={
        Object {
          "margin": "0 auto",
          "width": "100%",
        }
      }
    >
      <div
        className="col-xs-12"
        style={
          Object {
            "marginBottom": "50px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="row center-xs"
          >
            <h2
              className="bp3-heading bp3-text-muted"
            >
              Deskreen Screen Viewer
            </h2>
          </div>
          <div
            className="row center-xs"
            style={
              Object {
                "margin": "0 auto",
                "width": "100%",
              }
            }
          >
            <div
              className="col-md-6 col-xl-4"
            >
              <div
                className="bp3-card bp3-elevation-3"
                style={
                  Object {
                    "backgroundColor": "rgba(240, 248, 250, 1)",
                    "marginBottom": "30px",
                  }
                }
              >
                <h3
                  className="bp3-heading"
                >
                  My Device Info:
                </h3>
                <div
                  className="bp3-callout"
                >
                  <div
                    className=""
                  >
                    Device Type: 
                  </div>
                  <span
                    className="bp3-popover-wrapper"
                  >
                    <span
                      className="bp3-popover-target"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <div
                        className=""
                        style={
                          Object {
                            "backgroundColor": "#00f99273",
                            "borderRadius": "20px",
                            "fontWeight": 900,
                            "paddingLeft": "10px",
                            "paddingRight": "10px",
                          }
                        }
                        tabIndex={0}
                      >
                        <div
                          className=""
                        >
                          Device IP: 
                        </div>
                      </div>
                    </span>
                  </span>
                  <div
                    className=""
                  >
                    Device Browser: 
                  </div>
                  <div
                    className=""
                  >
                    Device OS: 
                  </div>
                </div>
                <div
                  className="bp3-text-muted"
                >
                  These details should match with the ones that you see in alert popup on computer screen, where Deskreen is running
                </div>
              </div>
            </div>
          </div>
          <div
            id="prompt-text"
            style={
              Object {
                "fontSize": "20px",
              }
            }
          >
            <h3
              className="bp3-heading"
            >
              Waiting for user to select source to share from screen sharing device...
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row top-xs"
    id="connecting-screen"
    style={
      Object {
        "height": "50vh",
        "marginLeft": "0px",
        "marginRight": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row center-xs top-xs"
      style={
        Object {
          "height": "100%",
          "marginLeft": "0px",
          "marginRight": "0px",
          "width": "100%",
        }
      }
    >
      <div
        className=""
      >
        <div
          className="row center-xs"
          style={
            Object {
              "height": "50px",
              "width": "100%",
            }
          }
        >
          <div
            className="col-xs-8 col-md-4"
          >
            <div
              className="css-9ebb9a"
            >
              <div
                className="css-1nih24x"
              />
              <div
                className="css-182ruc5"
              />
              <div
                className="css-fapzgm"
              />
              <div
                className="css-fqj3wr"
              />
              <div
                className="css-srr2wk"
              />
              <div
                className="css-1netspe"
              />
            </div>
          </div>
        </div>
        <div
          className="row center-xs"
        >
          <div
            className=""
          >
            <div
              className="react-reveal"
              style={
                Object {
                  "opacity": undefined,
                }
              }
            >
              <span
                className="bp3-icon bp3-icon-application"
                icon="application"
              >
                <svg
                  data-icon="application"
                  fill="#5C7080"
                  height={60}
                  viewBox="0 0 20 20"
                  width={60}
                >
                  <desc>
                    application
                  </desc>
                  <path
                    d="M3.5 9h9c.28 0 .5-.22.5-.5s-.22-.5-.5-.5h-9c-.28 0-.5.22-.5.5s.22.5.5.5zm0 2h5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5h-5c-.28 0-.5.22-.5.5s.22.5.5.5zM19 1H1c-.55 0-1 .45-1 1v16c0 .55.45 1 1 1h18c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm-1 16H2V6h16v11zM3.5 13h7c.28 0 .5-.22.5-.5s-.22-.5-.5-.5h-7c-.28 0-.5.22-.5.5s.22.5.5.5z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
